src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -902,73 +900,63 @@ void AdvertiserMinMdns::AdvertiseRecords(BroadcastAdvertiseType type)
 
     while (allInterfaces->Next(&interfaceId, &addressType))
     {
-        UniquePtr<IpAddressIterator> allIps = GetAddressPolicy()->GetIpAddressesForEndpoint(interfaceId, addressType);
-        VerifyOrDieWithMsg(allIps != nullptr, Discovery, "Failed to allocate memory for ip addresses.");
+        chip::Inet::IPPacketInfo packetInfo;
 
-        Inet::IPAddress ipAddress;
-        while (allIps->Next(ipAddress))
-        {
-            if (!ShouldAdvertiseOn(interfaceId, ipAddress))
-            {
-                continue;
-            }
+        packetInfo.Clear();
 
-            chip::Inet::IPPacketInfo packetInfo;
+        // Attempt to get a "Real" IP address since `Type()` on a broadcast address
+        // always returns IPAddressType::kAny
+        CHIP_ERROR err = GetFirstIpAddress(interfaceId, addressType, packetInfo.SrcAddress);
 
-            packetInfo.Clear();
-            packetInfo.SrcAddress = ipAddress;
```

Comment:
So we used to send packets for each of the addresses on the interface, right?  Now we just send them for the first address that matches the given address type.

If the interface has both a link-local address and a non-link-local address that seems like a behavior change.  Is the new behavior still correct?

Reply:
I tested it and while it seemed correct on my linux, OpenIOT SDK says it was not ... so reverted back to "broadcast sends over every interface explicitly".

Reply:
For the address bit though: it seems correct to not try to send for every address as the send cares about interface only ... the loop by address just made the broadcast needlesly multiply packets.

Reply:
So the SrcAddress is not actually relevant for these packets?

Reply:
Added a longer comment explaining.

We just need some IP address of a correct type. I believe even 127.0.0.1 or ::1 may work (unless they get mapped to kAny ... did not try) however getting a real address seems slightly more sane.

More work would be to pass in a type for broadcast and not an address, so the broadcast destination would be `oneof(IPAddressForUnicast, InterfaceIdForBroadcast)` however that change sounds larger.